source/OptimaJet.DWKit.StarterApplication/Repositories/ProjectRepository.cs

```diff
@@ -48,35 +54,104 @@ public override IQueryable<Project> Get()
 
         public override async Task<Project> UpdateAsync(int id, Project entity)
         {
-            ValidateOrganization(entity);
+            var result = ValidateOranizationUpdate(entity, id);
             return await base.UpdateAsync(id, entity);
         }
 
         public override async Task<Project> CreateAsync(Project entity)
         {
-            ValidateOrganization(entity);
+            ValidateOrganizationAdd(entity);
             return await base.CreateAsync(entity);
         }
 
-        private void ValidateOrganization(Project project)
+
+        private bool ValidateOranizationUpdate(Project projectUpdate, int id)
```

Comment:
Instead of writing validation here, in the repository, I think the Form object pattern would suit this better. The form objects would then be used form the service, rather than the repository. The reason for this is that the repository should care as little about application/domain logic as possible. 

For errors, you'll also probably want to utilize jsonapi-errors rather than logging.

Some resources:

https://json-api-dotnet.github.io/#/custom-errors

I believe, according to this documentation, if you raise an exception, the jsonapi middleware will correctly capture it for you. (be sure to have tests that verify this though).
- this would clean up the code _a lot_ :)